src/box/replication.cc

```diff
@@ -930,13 +942,33 @@ struct replicaset_connect_state {
 	int failed;
 	/** Signaled when an applier connects or stops. */
 	struct fiber_cond wakeup;
+	/** An applier which connected to the bootstrap leader. */
+	struct applier *bootstrap_leader_applier;
```

Comment:
I would recommend to reconsider if you really need this as a member. Maybe better find the leader applier by a simple fullscan of all the appliers when needed. Then you won't have to care about updating this member. IMO it is not needed often enough to keep it cached and complicate this struct.